jormungandr/src/rest/v0/handlers.rs

```diff
@@ -19,14 +19,15 @@ use bytes::{Bytes, IntoBuf};
 use futures::{future, Future, IntoFuture, Stream};
 use std::convert::Infallible;
 use std::str::FromStr;
+use std::sync::Arc;
 
 pub use crate::rest::Context;
 
 macro_rules! ActixFuture {
     () => { impl Future<Item = impl Responder + 'static, Error = impl Into<Error> + 'static> + 'static }
 }
 
-fn chain_tip_fut<'a>(context: &State<Context>) -> impl Future<Item = Ref, Error = Error> {
+fn chain_tip_fut<'a>(context: &State<Context>) -> impl Future<Item = Arc<Ref>, Error = Error> {
```

Comment:
Maybe `Ref` could encapsulate the `Arc` under the hood, but maybe explicit is better.

Reply:
sure that would make sense. there are pro and cons to encapsulate and hide the `Arc`.